hathor/p2p/node_sync.py

```diff
@@ -741,8 +745,12 @@ def on_tx_success(self, tx: 'BaseTransaction') -> 'BaseTransaction':
         # the parameter of the second callback is the return of the first
         # so I need to return the same tx to guarantee that all peers will receive it
         if tx:
-            # Add tx to the DAG.
-            success = self.manager.on_new_tx(tx)
+            assert tx.hash is not None
+            if self.manager.tx_storage.transaction_exists(tx.hash):
```

Comment:
This wasn't removed from `on_new_tx`, `on_new_tx` considers it a "soft error" if the tx already exists (it raises an exception unless `fails_silently=True`. But this case is perfectly normal and common, so I think it's best to just not call `on_new_tx` when it's perfectly fine for a "duplicate" tx to be received. Otherwise it will keep logging a bunch of warnings that are completely useless.

Reply:
I got it. Nice clean up in the logs.